From c260cd124d5990395897637ed8bd5edb2fbe7ac6 Mon Sep 17 00:00:00 2001
From: Albert Uchytil <auchytil@dhcp-24-234.brq.redhat.com>
Date: Mon, 3 Mar 2014 08:41:14 +0100
Subject: [PATCH] Added Provides: scl-package(%scl) for metapackage and build
 subpackage

---
 macros.scl | 1 +
 scl.attr   | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/macros.scl b/macros.scl
index 9e4b0540ed75617346334816abca53f8991dece4..50f4df2e6ac09ae1de7cde5e49f7c4c82db6ef70 100644
--- a/macros.scl
+++ b/macros.scl
@@ -73,6 +73,7 @@ package or when debugging this package.
 Requires: %scl_runtime
 }
 BuildRequires: scl-utils-build
+Provides: scl-package(%scl)
 %{?scl_package_override:%scl_package_override}
 }
 
diff --git a/scl.attr b/scl.attr
index 3e546568261b996e065be3bfaed648d785466c36..5f7d91f452b8ba1e29e98892d0e0145fae89e55e 100644
--- a/scl.attr
+++ b/scl.attr
@@ -1,2 +1,2 @@
 %__scl_provides	%{_rpmconfigdir}/scldeps.sh --provides %{scl}
-%__scl_path	%{?scl:^%{_scl_prefix}/.*$}
+%__scl_path	%{?scl:^%{_scl_prefix}/.*$ || %{_root_sysconfdir}/rpm/macros.%{scl}-config$}
--
1.8.3.1

