From f6d35d3406cf57024d3fe44d308d7e11527e3551 Mon Sep 17 00:00:00 2001
From: Albert Uchytil <auchytil@redhat.com>
Date: Wed, 2 Apr 2014 17:01:41 +0200
Subject: [PATCH] Added conditional dependencies for main metapackage

---
 macros.scl | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/macros.scl b/macros.scl
index ac83ae302ce971c4564c107bd0a1c769db6465e2..cd19c593002eb1c823a12b24df43196374086120 100644
--- a/macros.scl
+++ b/macros.scl
@@ -53,10 +53,12 @@
     /usr/lib/rpm/redhat/brp-python-hardlink
     %{!?__jar_repack:/usr/lib/rpm/redhat/brp-java-repack-jars}
 %{nil}}
-Requires: %scl_runtime
 }
 BuildRequires: scl-utils-build
+%if "%{?scl}%{!?scl:0}" == "%{pkg_name}"
 Provides: scl-package(%scl)
+Requires: %scl_runtime
+%endif
 %{?scl_package_override:%scl_package_override}
 }
 
-- 
1.8.5.3

