diff -up scl-utils-2.0.1/rpm/macros.scl.repack-jars scl-utils-2.0.1/rpm/macros.scl
--- scl-utils-2.0.1/rpm/macros.scl.repack-jars	2017-03-20 09:16:51.194712064 +0200
+++ scl-utils-2.0.1/rpm/macros.scl	2017-03-20 09:18:25.558540724 +0200
@@ -94,7 +94,6 @@ package or when debugging this package.
     /usr/lib/rpm/brp-strip-static-archive %{__strip}
     /usr/lib/rpm/brp-scl-python-bytecompile %{__python} %{?_python_bytecompile_errors_terminate_build} %{_scl_root}
     /usr/lib/rpm/brp-python-hardlink
-    %{!?__jar_repack:/usr/lib/rpm/redhat/brp-java-repack-jars}
 %{nil}}
 BuildRequires: scl-utils-build
 %if "%{?scl}%{!?scl:0}" == "%{pkg_name}"
